client/qiskit_serverless/utils/runtime_service_client.py

```diff
@@ -0,0 +1,120 @@
+======================================================================
+
+.. currentmodule:: qiskit_serverless.utils.runtime_service_client
+
+Qiskit Serverless runtime client wrapper
+========================================
+
+.. autosummary::
+    :toctree: ../stubs/
+
+    ServerlessRuntimeService
+"""
+import os
+import logging
+from typing import Callable, Dict, Sequence, Type, Union, Optional
+
+import requests
+from qiskit_ibm_runtime import QiskitRuntimeService
+from qiskit_ibm_runtime.runtime_job import RuntimeJob
+from qiskit_ibm_runtime.runtime_job_v2 import RuntimeJobV2
+from qiskit_ibm_runtime.runtime_options import RuntimeOptions
+from qiskit_ibm_runtime.utils.result_decoder import ResultDecoder
+
+from qiskit_serverless.core.constants import (
+    REQUESTS_TIMEOUT,
+    ENV_JOB_GATEWAY_TOKEN,
+    ENV_JOB_GATEWAY_HOST,
+    ENV_JOB_ID_GATEWAY,
+    ENV_GATEWAY_PROVIDER_VERSION,
+    GATEWAY_PROVIDER_VERSION_DEFAULT,
+)
+
+
+def associate_runtime_job_with_serverless_job(
+    runtime_job_id: str, session_id: Optional[str] = None
+) -> bool:
+    """Make a request to gateway to associate runtime job id with serverless job id.
+
+    Args:
+        runtime_job_id (str): job id for runtime primitive
+        session_id (str): session/batch id
+
+    Returns:
+        bool: if request was ok
+    """
+    version = os.environ.get(ENV_GATEWAY_PROVIDER_VERSION)
+    if version is None:
+        version = GATEWAY_PROVIDER_VERSION_DEFAULT
+
+    token = os.environ.get(ENV_JOB_GATEWAY_TOKEN)
+    if token is None:
+        logging.warning("Runtime job will not be associated with serverless job.")
+        return False
+
+    url = (
+        f"{os.environ.get(ENV_JOB_GATEWAY_HOST)}/"
+        f"api/{version}/jobs/{os.environ.get(ENV_JOB_ID_GATEWAY)}/add_runtimejob/"
+    )
+    response = requests.post(
+        url,
+        data={"runtime_job": runtime_job_id, "session_id": session_id},
+        headers={"Authorization": f"Bearer {token}"},
+        timeout=REQUESTS_TIMEOUT,
+    )
+    if not response.ok:
+        sanitized = response.text.replace("\n", "").replace("\r", "")
+        logging.warning("Something went wrong: %s", sanitized)
+
+    return response.ok
+
+
+class ServerlessRuntimeService(QiskitRuntimeService):
+    """Serverless wrapper for QiskitRuntimeService.
+
+    Used for associating runtime jobs with serverless jobs.
+
+    Args:
+        QiskitRuntimeService (QiskitRuntimeService): Qiskit runtime service object.
+    """
+
+    def run(
+        self,
+        program_id: str,
+        inputs: Dict,
+        options: Optional[Union[RuntimeOptions, Dict]] = None,
+        callback: Optional[Callable] = None,
+        result_decoder: Optional[
+            Union[Type[ResultDecoder], Sequence[Type[ResultDecoder]]]
+        ] = None,
+        session_id: Optional[str] = None,
+        start_session: Optional[bool] = False,
+    ) -> Union[RuntimeJob, RuntimeJobV2]:
+        runtime_job = super().run(
```

Comment:
Is this the only method that requests runtine job?

Reply:
it looks like